src/SMU-Keithley_2400/main.py

```diff
@@ -106,6 +117,43 @@ def get_GUIparameter(self, parameter={}):
         if self.average_gui > 100:
             self.average = 100
 
+        if self.sweepvalue == "List sweep" and self.listtype == "Sweep":
+            self.listsweep_start = float(parameter["ListSweepStart"])
+            self.listsweep_end = float(parameter["ListSweepEnd"])
+            self.listsweep_steppoints_type = parameter["ListSweepStepPointsType"]
+            self.listsweep_steppoints_value = float(parameter["ListSweepStepPointsValue"])
+            self.listsweep_dual = float(parameter["ListSweepDual"])
+
+        if self.sweepvalue == "List sweep" and self.listtype == "Custom":
+            self.custom_values = str(parameter["ListSweepCustomValues"])
+
+        # source delay, time between applying source value and measurement
+        # use None (later AUTO), if empty 
+        try:
+            self.listsweep_hold = float(parameter["ListSweepHoldtime"])
+        except:
+            self.listsweep_hold = None
+
+        # trigger delay, time between trigger and applying source value
+        # use None (later AUTO), if empty
+        try:
+            self.listsweep_delay = float(parameter["ListSweepDelaytime"])
+        except:
+            self.listsweep_delay = None
+
+        # in case of a list sweep, the time stamp of each measurement point must be saved additionally
+        # to every (V,I) tuple
+        if self.sweepvalue == "List sweep":
+            self.variables = ["Voltage", "Current", "Timestamp"]
```

Comment:
Does the device really return a timestamp? From the unit `s` I guess it's rather the (elapsed) time?

Reply:
It is right that this is rather some time since the start of the last instrument reset or since starting the instrument that indicates when the measurement was done and it is a plotable number. One could also call it just "Time", but "Timestamp" is also not completely incorrect as it is often the wording of the manufacturer. For example it is also named "relative timestamp" in the documentation.